inference-engine/src/cldnn_engine/cldnn_graph.cpp

```diff
@@ -252,34 +316,193 @@ InferenceEngine::ICNNNetwork::Ptr CLDNNGraph::GetExecGraphInfoByPrimitivesInfo(s
-        std::string exec_time = "not_executed";
-        if (perfMap.find(prim_info.original_id) != perfMap.end()) {
-            auto perfCounter = perfMap.at(prim_info.original_id).second;
-            if (perfCounter.num > 0) {
-                exec_time = std::to_string(perfCounter.realTime_avg());
+        return inputs;
+    };
+
+    auto create_ngraph_node = [&](const cldnn::primitive_info& prim_info) {
+        const auto& deps = prim_info.c_dependencies;
+        const auto& user_ids = prim_info.c_users;
+        size_t output_size = user_ids.size();
+        bool is_output = user_ids.empty();
+        auto desc = desc_from_layout(prim_info.output_layout);
+        std::shared_ptr<ngraph::Node> return_node;
+
+        if (prim_info.type_id == "input_layout") {
+            auto param = std::make_shared<ngraph::op::Parameter>(
+                details::convertPrecision(desc.getPrecision()),
+                ngraph::PartialShape(desc.getDims()));
+            params.push_back(param);
+            return_node = param;
+        } else {
+            return_node = std::make_shared<ExecGraphInfoSerialization::ExecutionNode>(
+                get_inputs(prim_info), output_size);
+
+            if (is_output) {    // create additinal result node
+                nodes.push_back(return_node);
+                node2layer[prim_info.original_id] = return_node;
+                return_node->set_output_type(0,
+                    details::convertPrecision(desc.getPrecision()),
+                    ngraph::PartialShape(desc.getDims()));
+                results.emplace_back(std::make_shared<ngraph::op::Result>(return_node->get_default_output()));
+            } else {
+                size_t port = 0;
+                for (auto& usr_id : user_ids) {
+                    auto usr_it = std::find_if(primitives_info.begin(),
+                                           primitives_info.end(),
+                                           [&](cldnn::primitive_info& entry) {
+                                                return entry.original_id == usr_id;
+                                            });
+                    if (usr_it == primitives_info.end())
+                        continue;
+
+                    return_node->set_output_type(port,
+                                                details::convertPrecision(desc.getPrecision()),
+                                                ngraph::PartialShape(desc.getDims()));
+                    port++;
+                }
             }
         }
 
-        layer->params[ExecGraphInfoSerialization::PERF_COUNTER] = exec_time;
-        layer->params[ExecGraphInfoSerialization::OUTPUT_LAYOUTS] = prim_info.layout_str;
-        layer->params[ExecGraphInfoSerialization::EXECUTION_ORDER] = std::to_string(prim_info.exec_id);
+        for (auto&& kvp : extract_primitive_metadata(prim_info)) {
+            if (kvp.first == "layerName") {
```

Comment:
Can't it be just read from prim_info without introducing any new property?

Reply:
Fixed.